src/meta/Base.ts

```diff
@@ -26,11 +24,18 @@ export class Base {
 		this._invalidating = global.requestAnimationFrame(this._invalidate);
 	}
 
-	protected requireNode(key: string): void {
-		this._requiredNodes.add(key);
-
-		if (!this.nodes.has(key)) {
-			this.invalidate();
+	protected requireNode(key: string, callback?: WidgetMetaRequiredNodeCallback): void {
+		const node = this.nodes.get(key);
+		if (node) {
+			callback && callback.call(this, node);
+		}
+		else {
+			const callbacks = this._requiredNodes.get(key) || [];
+			callback && callbacks.push(callback);
+			this._requiredNodes.set(key, callbacks);
+			if (!callback) {
+				this.invalidate();
```

Comment:
Oh! I get it! I just wrote up this big complicated example of why I thought we needed that invalidate when I realized that we don't 😝